src/Server.cpp

```diff
@@ -1,182 +1,188 @@
-        // Blocking call to epoll_wait
-        int	nfds = epoll_wait(m_epfd, events, MAX_EVENTS, -1);
-        if (nfds < 0)
-            throw std::runtime_error("epoll_wait");
-
-        for (int n = 0; n < nfds; ++n) {
-            if (events[n].data.fd == m_serverSock)
-                acceptConnection();
-            else
-                handleConnections(events[n].data.fd);
-        }
-    }
+void Server::run()
+{
+	while (true) {
+		struct epoll_event events[MAX_EVENTS]; // NOLINT
+		// Blocking call to epoll_wait
+		int nfds = epoll_wait(m_epfd, events, MAX_EVENTS, -1);
+		if (nfds < 0)
+			throw std::runtime_error("epoll_wait");
+
+		for (int i = 0; i < nfds; ++i) {
+			if (events[i].data.fd == m_serverSock)
+				acceptConnection();
+			else
+				handleConnections(events[i].data.fd);
+		}
+	}
 }
 
 /**
  * @brief Accept new connections
- * 
- * A new connection is accepted, its address and port saved in _cliendAddr and a 
- * file descriptor for the new client socket is saved in clientSock. This 
- * information is updated each time a new connection is accepted. The 
+ *
+ * A new connection is accepted, its address and port saved in _cliendAddr and a
+ * file descriptor for the new client socket is saved in clientSock. This
+ * information is updated each time a new connection is accepted. The
  * clientSock is added to the list of fds watched by epoll.
- * 
+ *
  * The while-loop allows for multiple connections to be accepted in one call
- * of this function. It is broken when accept sets errno to EAGAIN or 
- * EWOULDBLOCK (equivalent error codes indicating that a non-blocking operation 
+ * of this function. It is broken when accept sets errno to EAGAIN or
+ * EWOULDBLOCK (equivalent error codes indicating that a non-blocking operation
  * would normally block), meaning that no more connections are pending.
  */
-void    Server::acceptConnection(){
-    while (1) {
-		struct sockaddr_in	clientAddr;
-        socklen_t			addr_len = sizeof(clientAddr);
-        int	clientSock = accept(
-            m_serverSock, (struct sockaddr*)&clientAddr, &addr_len);
-        if (clientSock < 0) {
-            if (errno == EAGAIN || errno == EWOULDBLOCK) {
-                break; // No more pending connections
-            } else {
-                std::cerr << "error: accept: " << strerror(errno) << "\n";
-                break;
-            }
-        }
-
-        // Add client socket to epoll instance
-		struct epoll_event	ev;
-        ev.events = EPOLLIN;
-        ev.data.fd = clientSock;
-        if (epoll_ctl(m_epfd, EPOLL_CTL_ADD, clientSock, &ev) < 0) {
-            std::cerr << "error: epoll_ctl: clientSock: " << strerror(errno) << "\n";
-            close(clientSock);
-        }
-    }
+void Server::acceptConnection()
+{
+	while (true) {
+		struct sockaddr_in clientAddr = {};
+		socklen_t addrLen = sizeof(clientAddr);
+		int clientSock = accept(m_serverSock, reinterpret_cast<struct sockaddr*>(&clientAddr), &addrLen); // NOLINT
```

Comment:
Yeah it doesn't like it. "Don't use "reinterpret_cast" - or in CPP guidelines ES.48: [Avoid casts](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es48-avoid-casts)

The problem is we have cast since the function interface is constructed this way.
Which can fall under ES.49: [If you must use a cast, use a named cast](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es49-if-you-must-use-a-cast-use-a-named-cast)